lib/ohai/plugins/linux/network.rb

```diff
@@ -406,17 +407,19 @@ def parse_ip_addr_inet6_line(cint, iface, line)
             ]
           end.first
 
-          unless route.nil? or route.empty?
+
+          if route && !route.empty?
+            macaddress iface[route[:dev]][:addresses].select{|k,v| v["family"]=="lladdr"}.first.first unless iface[route[:dev]][:flags].include? "NOARP"
```

Comment:
I think this is fine, but ftr, if you have v4 and v6 routes that match, v6 will win since it's done second.